File: ARCADE/Intro/Level 4_4: arrayChange.cs
    using System;
     
    public class Test
    {
    	public static int arrayChange(int[] inputArray){
            int result = 0;
            for (int i = 1; i < inputArray.Length; i++){
                if (inputArray[i] <= inputArray[i - 1]){
                    result += inputArray[i - 1] - inputArray[i] + 1;
                    inputArray[i] = inputArray[i - 1] + 1;
                }
            }
            return result;
    	}
    	public static void Main()
    	{
    		int [] inputArray = new int []{1, 1, 1};
    		Console.WriteLine(arrayChange(inputArray));
    	}
    }

File: ARCADE/Intro/Level 4_4: arrayChange.pas
program arrayChange;
 
Type
	ArrayInteger = Array of integer;
 
function arrayChange(inputArray: ArrayInteger): integer;
var i, count : integer;
begin
    count := 0;
    for i := 1 to length(inputArray) - 1 do
        if(inputArray[i] <= inputArray[i-1]) then
            begin
                count := count + inputArray[i-1] - inputArray[i] + 1;
                inputArray[i] := inputArray[i-1] + 1;
            end;
 
    arrayChange := count;
end;
 
 
var inputArray : ArrayInteger;
begin
	inputArray := ArrayInteger.Create(1, 1, 1);
	writeln(arrayChange(inputArray));
end.
